packages/@sanity/form-builder/src/__workshop__/example.tsx

```diff
@@ -0,0 +1,140 @@
+import {Card, Grid, Stack, useToast} from '@sanity/ui'
+import {useBoolean} from '@sanity/ui-workshop'
+import React, {useCallback, useEffect, useMemo, useState} from 'react'
+import {Patcher} from '@sanity/mutator'
+import type {SchemaType, ObjectField} from '@sanity/types'
+import {
+  FormBuilder,
+  FormBuilderInput,
+  PatchEvent,
+} from '../sanity/legacyPartImplementations/form-builder'
+import {applyAll} from '../patch/applyPatch'
+import {toGradient} from '../sanity/utils/gradientPatchAdapter'
+import {getDummySchema, getDummyDocument, DUMMY_DOCUMENT_ID} from './_common/data'
+import {FilterFieldInput, FormBuilderTester, FormDebugger} from './_common/input'
+
+const patchChannel = FormBuilder.createPatchChannel()
+
+export default function ExampleStory() {
+  const ref = React.useRef()
+  const toast = useToast()
+  const isUseMutator = useBoolean('Use Mutator', false, 'Props')
+  const [, setFocused] = useState(false)
+  const [focusPath, setFocusPath] = useState([])
+  const isReadOnly = useBoolean('Read-only', false, 'Props')
+  const isFilterFields = useBoolean('Filter Fields', false, 'Props')
+  const isHiddenGroup = useBoolean('Hidden Group', false, 'Props')
+  const includeUnknownField = useBoolean('Unknown Field in Value', false, 'Props')
+  const isDebug = useBoolean('Debug', false, 'Props')
+  const [documentValue, setDocumentValue] = useState<{[key: string]: any}>(getDummyDocument())
+  const [_fieldFilter, setFieldFilter] = useState<string>(``)
+  const EMPTY = []
+
+  const schema = useMemo(() => {
+    return getDummySchema({
+      hiddenGroup: isHiddenGroup,
+    })
+  }, [isHiddenGroup])
+  const documentType = useMemo(() => {
+    return schema.get('book')
+  }, [schema])
+
+  const handleChange = useCallback((patchEvent: PatchEvent) => {
+    setDocumentValue((currentDocumentValue) => applyAll(currentDocumentValue, patchEvent.patches))
+  }, [])
+  const handleChangeMutator = useCallback((patchEvent: PatchEvent) => {
+    const patcher = new Patcher(
+      toGradient(patchEvent.patches).map((patch) => ({...patch, id: DUMMY_DOCUMENT_ID}))
+    )
+    setDocumentValue((currentDocumentValue) => patcher.apply(currentDocumentValue))
+  }, [])
+  const handleBlur = useCallback(() => setFocused(false), [])
+  const handleFocus = useCallback((path) => {
+    setFocusPath(path)
+    setFocused(true)
+  }, [])
+  const handleChangeFieldFilter = useCallback((value) => {
+    const handledValue = value && value.length > 0 ? value : ``
+
+    setFieldFilter(handledValue)
+    toast.push({status: 'success', title: `Updated field filter`})
+  }, [])
+  const fieldFilter = useCallback(
```

Comment:
this will parse the function source every time we filter a field. Would it be a possible improvement/simplification to parse it in a useMemo that depends on the fieldFilter source string?

Reply:
Memoized the filter 👍